### PR TITLE
BS-4056: Add pdp.interceptAllEndpoints, pdp.ignoreEndpoints, pdp.ignoreRegex

### PR DESCRIPTION
**Pull request by yashtewari**
New configuration options added.

Also in this PR: BS-4065

### CODE REVIEW

File: src/main/java/security/build/pdp/client/PDPInterceptor.java [hunk 1]
```diff
@@ -29,6 +34,18 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
...
         }
 
+	    for (String e: ignoreEndpoints) {
+	        if (e == request.getRequestURI()) {
+	            return true;
+            }
+        }
+
+	    for (String r: ignoreRegex) {
+	        if (Pattern.compile(r).matcher(request.getRequestURI()).matches()) {
```
Comment (eyalkraft) [inline src/main/java/security/build/pdp/client/PDPInterceptor.java:44]: this looks like all the regex patterns will be recompiled on each request. maybe the compilation of the patterns can happen in the setter? do values that originate in the configuration go through the setter?

File: src/main/java/security/build/pdp/client/PDPInterceptor.java [hunk 2]
```diff
@@ -29,6 +34,18 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
 	        return true;
         }
 
+	    for (String e: ignoreEndpoints) {
+	        if (e == request.getRequestURI()) {
```
Comment (eyalkraft) [inline src/main/java/security/build/pdp/client/PDPInterceptor.java:38]: is there a `contains` alternative?

File: src/test/java/security/build/pdp/client/PDPInterceptorTest.java [hunk 3]
```diff
@@ -74,4 +77,52 @@ void PreHandle_Disabled_Authorized() throws Throwable {
         Boolean isAuthorized = pdpInterceptor.preHandle(null, response, handlerMethod);
         Assertions.assertEquals(true, isAuthorized);
     }
+
```
Comment (eyalkraft) [inline src/test/java/security/build/pdp/client/PDPInterceptorTest.java:80]: 👏

File: src/main/java/security/build/pdp/client/PDPInterceptor.java [hunk 4]
```diff
@@ -18,6 +19,10 @@
 
     @Value("${pdp.enable:true}")
     private boolean enable;
+    @Value("${pdp.ignoreEndpoints:}")
```
Comment (eyalkraft) [inline src/main/java/security/build/pdp/client/PDPInterceptor.java:22]: please add description to the README

**Review by eyalkraft — CHANGES_REQUESTED** [review]

File: src/main/java/security/build/pdp/client/PDPInterceptor.java (hunk 2 above)
Comment (yashtewari) [reply src/main/java/security/build/pdp/client/PDPInterceptor.java]: Not for array, we could use Lists or something.

**Review by eyalkraft — APPROVED** [review]
please merge for GW to use today and let's take care of the comments in a following PR (please leave a comment with the task number)

**Comment by yashtewari** [comment]
Suggestions on this are now part of `BS-4066`

**Pull request merged**